psydac/linalg/tests/test_linalg.py

```diff
@@ -33,6 +31,47 @@ def compute_global_starts_ends(domain_decomposition, npts):
 
     return global_starts, global_ends
 
+def get_StencilVectorSpace(n1, n2, p1, p2, P1, P2):
+    npts = [n1, n2]
+    pads = [p1, p2]
+    periods = [P1, P2]
+    D = DomainDecomposition(npts, periods=periods)
+    global_starts, global_ends = compute_global_starts_ends(D, npts)
+    C = CartDecomposition(D, npts, global_starts, global_ends, pads=pads, shifts=[1,1])
+    V = StencilVectorSpace(C)
+    return V
+
+def get_positive_definite_nparray(n1, n2):
+    np.random.seed(2)
+    mat = np.zeros((n1*n2, n1*n2))
+    for i in range(1, n1):
+        R = np.random.random((n2, n2))
+        for j in range(n1-i):
+            mat[j*n2:(j+1)*n2, (i+j)*n2:(i+j+1)*n2] = R
+    mat += np.transpose(mat)
+    sums = [sum(mat[i, :]) for i in range(n1*n2)]
+    max_sums = [max([sums[i*n2+j] for i in range(n1)]) for j in range(n2)]
+    D = np.random.random((n2, n2))
+    D += np.transpose(D)
+    for i in range(n2):
+        z = [D[i, (i+j)%n2] for j in range(1, n2)]
+        D[i, i] *= (max_sums[i] + sum(z)) / ( D[i, i] * np.random.random(1) )
+    for i in range(n1):
+        mat[i*n2:(i+1)*n2, i*n2:(i+1)*n2] = D
+    mat /= n1*n2 # to avoid exploding determinants
+    return mat
```

Comment:
1. I actually did not consider creating sparse matrices, that would certainly speed up the code. However, 
2. I took inspiration from test_stencil_matrix.py line 1395-1397:
   ```
   for k1 in range(-p1,p1+1):
        for k2 in range(-p2,p2+1):
            M[:,:,k1,k2] = ...
   ```
   That is to say, my method get_positive_definite_nparray creates a matrix that is of this specific tensor-product banded structure.

   In particular, in line 50 I fill each upper block-co-diagonal with the same random matrix R. In line 51 I fill each lower block-co-diagonal in a symmetric way. In line 54 I create what will be the block-diagonal matrix, in line 55 I make it symmetric, in line 58 I manually ensure that my matrix will be positive definite, in line 60 I fill the block-diagonal. While a bit hard to read, I am quite content with the method and the tests are not running that long as I chose relatively small values for n1 and n2. 
   Do you want me to use `scipy.sparse` anyways to create these test matrices for better readability?

Reply:
It is indeed a bit complicated. It seems to me that the code would be simpler if you used the multi-index notation of `StencilMatrix` objects, which are indexed with `[i1, i2, j1-i1, j2-i2]`, so that you only process the non-zero elements. The main diagonal would be `[:, :, 0, 0]`. After conversion to `numpy` or `scipy.sparse` format, you can check with those libraries if the resulting matrix is positive definite.

Reply:
7fc1cb2dfbbaad9fab59690f2a18f850489f4d30